src/sort/sort.cpp

```diff
@@ -164,8 +164,53 @@ void merge_sort(std::vector<uint16_t>& arr,
 }
 
 
-void quick_sort(std::vector<uint16_t>& arr) {
-    std::sort(arr.begin(), arr.end()); // TODO: remove this and actually implement the algo.
+void quick_sort(std::vector<uint16_t>& arr, 
+                int start_idx,
+                int end_idx,
+                std::function<void(std::vector<uint16_t>,
+                                   std::unordered_map<uint16_t, std::string>)> display) {
+    if (start_idx < end_idx) {
+        // Find the pivot point at the center of the subarray
+        uint16_t pivot = arr[(start_idx + end_idx) / 2];
+        int left_ptr = start_idx;
+        int right_ptr = end_idx;
+
+        std::unordered_map<uint16_t, std::string> color_map{{start_idx, YELLOW}, {end_idx, YELLOW}, {pivot, GREEN}};
+        display(arr, color_map);
+
+        while (left_ptr <= right_ptr) {
+            // Move left pointer to the right until a value smaller than the pivot is found
+            while (arr[left_ptr] < pivot) {
+                left_ptr++;
+                std::unordered_map<uint16_t, std::string> color_map{{start_idx, YELLOW}, {end_idx, YELLOW}, {left_ptr, RED}, {right_ptr, RED}, {pivot, GREEN}};
```

Comment:
Line is too long. Something like this is easier to read.
```suggestion
                std::unordered_map<uint16_t, std::string> color_map {
                    {start_idx, YELLOW}, {end_idx, YELLOW}, {left_ptr, RED},
                    {right_ptr, RED}, {pivot, GREEN}};
```